dm/worker/server.go

```diff
@@ -210,20 +212,30 @@ func (s *Server) setWorker(worker *Worker, needLock bool) {
 	s.worker = worker
 }
 
-func (s *Server) getWorkerErr(needLock bool) error {
+func (s *Server) getSourceStatus(needLock bool) pb.SourceStatus {
 	if needLock {
 		s.Lock()
 		defer s.Unlock()
 	}
-	return s.workerErr
+	return s.sourceStatus
 }
 
-func (s *Server) setWorkerErr(workerErr error, needLock bool) {
+func (s *Server) setSourceStatus(source string, err error, needLock bool) {
 	if needLock {
 		s.Lock()
 		defer s.Unlock()
 	}
-	s.workerErr = workerErr
+	s.sourceStatus = pb.SourceStatus{
```

Comment:
Do we never set `RelayStatus` for `s.sourceStatus`, right?

Reply:
Yes

Reply:
Add a comment to clarify it?

Reply:
addressed in https://github.com/pingcap/dm/pull/490/commits/b9f4793fa1b4d18afa02c9fd9987ce42991dad44